include/tscore/ink_queue.h

```diff
@@ -57,10 +57,12 @@ void ink_queue_load_64(void *dst, void *src);
 #define INK_QUEUE_LD64(dst, src) (ink_queue_load_64((void *)&(dst), (void *)&(src)))
 #endif
 
+// passing a const volatile value of 0 works around a gcc bug
 #if TS_HAS_128BIT_CAS
 #define INK_QUEUE_LD(dst, src)                                                       \
+  const volatile __int128_t x = 0;                                                   \
   do {                                                                               \
```

Comment:
The purpose of the do/while(0) is that `INK_QUEUE_LD(x, y);`  should expand to a single statement.  So you wouldn't get a bad surprise if you used it in an if or while statement:
```
if (test)
  INK_QUEUE_LD(x, y);
```
In any case, should we replace the builtin with one that isn't obsolete? https://gcc.gnu.org/onlinedocs/gcc/_005f_005fatomic-Builtins.html